### PR TITLE
AtlasTmxMapLoader: Supports group layers now

### PR DESCRIPTION
**Pull request by TCreutzenberg**
AtlasTmxMapLoader: Supports groups now

Can be checked in TiledMapAtlasAssetManagerTest, test.tmx has been adjusted to contain a group layer now.

### CODE REVIEW

**Comment by TCreutzenberg** [comment]
Hey guys!

This pull request is open for a month, what do I have to do to have this rejected or merged?
Did I do something wrong? Please tell me so I can correct things.

Thank you very much in advance.

**Comment by NathanSweet** [comment]
Sorry, sometimes libgdx maintainers are busy, plus it is summertime. I don't personally use the tile map stuff, so I'm not familiar with it at all and don't usually take responsibility to evaluate, test, merge, and maintain that portion of libgdx. I'm sure someone will be along shortly to check this out. You could try pinging them using @username (check commit logs for who usually fields related PRs).

**Comment by TCreutzenberg** [comment]
@MobiDevelop @manuelbua 
Hey guys, how are you?
As you are stated as the authors of the AtlasTmxMapLoader, can you please check this pull request?
Thank you very much in advance!

**Comment by manuelbua** [comment]
Hi @TCreutzenberg, unfortunately i'm not involved in libgdx/gamedev anymore, not sure @MobiDevelop still working on it?

File: gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java [hunk 1]
```diff
@@ -252,20 +255,49 @@ protected TiledMap loadMap (Element root, FileHandle tmxFile, AtlasResolver reso
...
+			loadLayer(map, map.getLayers(), element, tmxFile, resolver);
 		}
 		return map;
 	}
 
+    protected void loadLayer(TiledMap map, MapLayers parentLayers, Element element, FileHandle tmxFile, AtlasResolver resolver) {
+        String elementName = element.getName();
+        if (elementName.equals("group")) {
+            loadTileGroup(map, parentLayers, element, tmxFile, resolver);
+        } else if (elementName.equals("properties")) {
```
Comment (MobiDevelop) [inline gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: A bit of a nitpick but "properties" isn't a layer, so having it be read in the `loadLayer` method seems incorrect.

**Comment by MobiDevelop** [comment]
Looking at the (*)TmxMapLoader classes, it really feels like there's an opportunity to refactor to put more into the base class. There's both code duplication and features that have been added to TmxMapLoader that AtlasTmxMapLoader hasn't received (likely due to the duplication).

File: gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java (hunk 1 above)
Comment (TCreutzenberg) [reply gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: You're right. The reason why I didn't change it is to kep it consistent with BaseTmxMapLoader.loadLayer(...) where nearly the same stuff happens.

Comment (MobiDevelop) [reply gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: `BaseTmxMapLayer#loadLayer` doesn't handle properties.

Comment (TCreutzenberg) [reply gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: So, what would you suggest? loadLayerAndProperties?

Comment (MobiDevelop) [reply gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: I'll just spend some time refactoring this weekend to get into a more sane place as far as duplication goes.

Comment (TCreutzenberg) [reply gdx/src/com/badlogic/gdx/maps/tiled/AtlasTmxMapLoader.java]: That's great! Thanks a lot!

**Comment by TCreutzenberg** [comment]
I will close this pull request as I have created a new one: https://github.com/libgdx/libgdx/pull/5495/

**Pull request closed without merging**